services/src/main/scala/cromwell/services/metadata/impl/ReadMetadataRegulatorActor.scala

```diff
@@ -4,44 +4,52 @@ import java.util.UUID
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props}
 import cromwell.core.Dispatcher.ApiDispatcher
+import cromwell.core.WorkflowId
 import cromwell.services.MetadataJsonResponse
 import cromwell.services.metadata.MetadataService
-import cromwell.services.metadata.MetadataService.{MetadataQueryResponse, BuildMetadataJsonAction, MetadataServiceAction, MetadataServiceResponse, RootAndSubworkflowLabelsLookupResponse, BuildWorkflowMetadataJsonAction}
+import cromwell.services.metadata.MetadataService.{BuildMetadataJsonAction, BuildWorkflowMetadataJsonAction, MetadataQueryResponse, MetadataServiceAction, MetadataServiceResponse, RootAndSubworkflowLabelsLookupResponse}
 import cromwell.services.metadata.impl.ReadMetadataRegulatorActor.PropsMaker
 import cromwell.services.metadata.impl.builder.MetadataBuilderActor
 
 import scala.collection.mutable
 
 class ReadMetadataRegulatorActor(metadataBuilderActorProps: PropsMaker, readMetadataWorkerProps: PropsMaker) extends Actor with ActorLogging {
   // This actor tracks all requests coming in from the API service and spins up new builders as needed to service them.
-  // If the processing of an identical request is already in flight the requester will be added to a set of requesters
+  // If the processing of an identical request is already in flight the requester will be added to a list of requesters
   // to notify when the response from the first request becomes available.
+  // Note that if someone sends the same action twice, this actor must reply twice too.
 
   // Map from requests (MetadataServiceActions) to requesters.
-  val apiRequests = new mutable.HashMap[MetadataServiceAction, Set[ActorRef]]()
+  val apiRequests = new mutable.HashMap[MetadataServiceAction, List[ActorRef]]()
```

Comment:
List appending can be expensive in scala when lists get long. It's unlikely in this situation but I would recommend using `Vector` instead in this situation

Reply:
I'm prepending to a `List`, not appending. So it should not be that expensive. Also I chose `List` because of it's better characteristics in case of small collection size (< 10 elements). I mistakenly assumed that the size of this `List` would not be very large. I will change it to `Vector` and notify you when I'm done.

Reply:
I didn't spot that it was prepending, and I suspect these will indeed be very short. Since you've properly considered it and decided that `List` is appropriate, 👍  from me to leaving it as a `List`